jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/Agent.java

```diff
@@ -0,0 +1,126 @@
+import io.opentelemetry.sdk.metrics.view.InstrumentSelector;
+import io.opentelemetry.sdk.metrics.view.View;
+import io.opentelemetry.sdk.resources.Resource;
+import jdk.jfr.EventSettings;
+import jdk.jfr.consumer.RecordingStream;
+import io.opentelemetry.contrib.jfr.metrics.RecordedEventHandler;
+import io.opentelemetry.contrib.jfr.metrics.HandlerRegistry;
+
+import java.lang.instrument.Instrumentation;
+import java.time.Duration;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static io.opentelemetry.sdk.metrics.common.InstrumentType.*;
+import static io.opentelemetry.sdk.metrics.data.AggregationTemporality.DELTA;
+
+public class Agent {
+    private static final Logger logger = Logger.getLogger(Agent.class.getName());
+
+    private static final String OTLP_URL = "OTLP_URL";
+    private static final String API_KEY = "API_KEY";
+    private static final String SERVICE_NAME = "SERVICE_NAME";
+    private static final String API_KEY_HEADER = "api-key";
+    private static final String SERVICE_NAME_HEADER = "service.name";
+    private static final String DEFAULT_URL = "http://localhost:4317";
+    private static final String DEFAULT_SERVICE_NAME = "jfr-otlp-bridge";
+    private static final long EXPORT_MILLIS = 2000;
+
+    private static MeterProvider meterProvider;
+
+    public static void agentmain(String agentArgs, Instrumentation inst) {
+      premain(agentArgs, inst);
+    }
+
+    public static void premain(String agentArgs, Instrumentation inst) {
+      meterProvider = GlobalMeterProvider.get();
+      if (meterProvider == null) {
+        configureOpenTelemetry();
+      }
+      startJfrService(agentArgs, inst);
+    }
+
+    public static void startJfrService(String agentArgs, Instrumentation inst) {
+        var jfrMonitorService = Executors.newSingleThreadExecutor();
+        var toMetricRegistry = HandlerRegistry.createDefault(meterProvider);
+
+        jfrMonitorService.submit(() -> {
+            try (var recordingStream = new RecordingStream()) {
+                var enableMappedEvent = eventEnablerFor(recordingStream);
+                toMetricRegistry.all().forEach(enableMappedEvent);
+                recordingStream.setReuse(false);
+                logger.log(Level.FINE, "Starting recording stream...");
+                recordingStream.start(); //run forever
+            }
+        });
+
+    }
+
+    private static Consumer<RecordedEventHandler> eventEnablerFor(RecordingStream recordingStream) {
+        return handler -> {
+            EventSettings eventSettings = recordingStream.enable(handler.getEventName());
+            handler.getPollingDuration().ifPresent(eventSettings::withPeriod);
+            handler.getThreshold().ifPresent(eventSettings::withThreshold);
+            recordingStream.onEvent(handler.getEventName(), handler);
+        };
+    }
+
+    static void configureOpenTelemetry() {
```

Comment:
You should use `sdk-extension-autoconfigure` to allow configuring OTel with our standard variables instead of configuring manually (you can use a meter provider configurer to add the aggregators)

Reply:
+1, was going to make the same comment.  As much as possible, we should rely on autoconfigure and push users to a common configuration component.

Again, if there were exposed as a library, you could assume OTel is initialized prior to wiring in your instrumentation (and accept a MeterProvider as your input).

Reply:
Thanks. Is there an example somewhere I can crib from?